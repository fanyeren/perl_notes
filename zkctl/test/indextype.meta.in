#!/bin/bash
source $(dirname $0)/common
TYPE=index-type
VALUE=dnews
raw='"category":"manual","deployment_next_partition":"1","deployment_policy":"full","layer":"layer1","partition_count":"20","replication_factor":"2","resource_requirement":"io-bound","trans_db_priority":"2"'

node_clear $TYPE

./zkctl -w $TYPE add --$TYPE-id=$VALUE --data='{'$raw'}'
./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
node_clear $TYPE
./zkctl -w $TYPE add --$TYPE-id=$VALUE --data={}
./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE add-meta-attr --$TYPE-id=$VALUE --key=${kv[0]} --value=${kv[1]}
    echo "Add meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -w $TYPE get-meta-attr --$TYPE-id=$VALUE --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
done

./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
echo "got all $TYPE result result=$?"

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE del-meta-attr --$TYPE-id=$VALUE --key=${kv[0]}
    echo "del meta attr ${kv[0]} result=$?"
done

./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
echo "got empty $TYPE result result=$?"

./zkctl -w $TYPE del --$TYPE-id=$VALUE
