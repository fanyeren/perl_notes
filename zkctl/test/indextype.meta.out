{"category":"manual","deployment_next_partition":"1","deployment_policy":"full","layer":"layer1","partition_count":"20","replication_factor":"2","resource_requirement":"io-bound","trans_db_priority":"2"}
{}
Add meta category=manual result 0
manual
Get meta category=manual result 0
Add meta deployment_next_partition=1 result 0
1
Get meta deployment_next_partition=1 result 0
Add meta deployment_policy=full result 0
full
Get meta deployment_policy=full result 0
Add meta layer=layer1 result 0
layer1
Get meta layer=layer1 result 0
Add meta partition_count=20 result 0
20
Get meta partition_count=20 result 0
Add meta replication_factor=2 result 0
2
Get meta replication_factor=2 result 0
Add meta resource_requirement=io-bound result 0
io-bound
Get meta resource_requirement=io-bound result 0
Add meta trans_db_priority=2 result 0
2
Get meta trans_db_priority=2 result 0
{"category":"manual","deployment_next_partition":"1","deployment_policy":"full","layer":"layer1","partition_count":"20","replication_factor":"2","resource_requirement":"io-bound","trans_db_priority":"2"}
got all index-type result result=0
del meta attr category result=0
del meta attr deployment_next_partition result=0
del meta attr deployment_policy result=0
del meta attr layer result=0
del meta attr partition_count result=0
del meta attr replication_factor result=0
del meta attr resource_requirement result=0
del meta attr trans_db_priority result=0
{}
got empty index-type result result=0
